src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3,16 +3,1430 @@
+
+        if (_settings->GlobalSettings().GetShowTabsInTitlebar())
+        {
+            // Remove the TabView from the page. We'll hang on to it, we need to
+            // put it in the titlebar.
+            uint32_t index = 0;
+            if (this->Root().Children().IndexOf(_tabRow, index))
+            {
+                this->Root().Children().RemoveAt(index);
+            }
+
+            // Inform the host that our titlebar content has changed.
+            _setTitleBarContentHandlers(*this, _tabRow);
+        }
+
+        //Event Bindings (Early)
+        _newTabButton.Click([this](auto&&, auto&&) {
+            this->_OpenNewTab(std::nullopt);
+        });
+        _tabView.SelectionChanged({ this, &TerminalPage::_OnTabSelectionChanged });
+        _tabView.Items().VectorChanged({ this, &TerminalPage::_OnTabItemsChanged });
+        _tabView.TabClosing({ this, &TerminalPage::_OnTabClosing });
+
+        _CreateNewTabFlyout();
+        _OpenNewTab(std::nullopt);
+
+        _tabContent.SizeChanged({ this, &TerminalPage::_OnContentSizeChanged });
+    }
+
+    // Method Description:
+    // - Show a ContentDialog with a single "Ok" button to dismiss. Looks up the
+    //   the title and text from our Resources using the provided keys.
+    // - Only one dialog can be visible at a time. If another dialog is visible
+    //   when this is called, nothing happens. See _ShowDialog for details
+    // Arguments:
+    // - titleKey: The key to use to lookup the title text from our resources.
+    // - contentKey: The key to use to lookup the content text from our resources.
+    void TerminalPage::ShowOkDialog(const winrt::hstring& titleKey,
+                                    const winrt::hstring& contentKey)
+    {
+        auto title = _resourceLoader.GetLocalizedString(titleKey);
+        auto message = _resourceLoader.GetLocalizedString(contentKey);
+        auto buttonText = _resourceLoader.GetLocalizedString(L"Ok");
+
+        Controls::ContentDialog dialog;
+        dialog.Title(winrt::box_value(title));
+        dialog.Content(winrt::box_value(message));
+        dialog.CloseButtonText(buttonText);
+
+        _showDialogHandlers(*this, dialog);
+    }
+
+    // Method Description:
+    // - Displays a dialog for errors found while loading or validating the
+    //   settings. Uses the resources under the provided  title and content keys
+    //   as the title and first content of the dialog, then also displays a
+    //   message for whatever exception was found while validating the settings.
+    // - Only one dialog can be visible at a time. If another dialog is visible
+    //   when this is called, nothing happens. See _ShowDialog for details
+    // Arguments:
+    // - titleKey: The key to use to lookup the title text from our resources.
+    // - contentKey: The key to use to lookup the content text from our resources.
+    void TerminalPage::ShowLoadErrorsDialog(const winrt::hstring& titleKey,
```

Comment:
Does this function and `ShowWarningsDialog` really need to live in `TerminalPage`? It seems to me like both these dialogs are only ever shown by the settings load process which happens exclusively in `App`.